src/Run.js

```diff
@@ -17,6 +18,11 @@ function Run(ref, json_q, callback, progressObj) {
       var image = drawarray[pos].image;
       var i = drawarray[pos].i;
       var input = ref.images[image].steps[i - 1].output;
+
+      getStep = function getStep(offset) {
```

Comment:
Hm, interesting! So it returns a /copy/ of the step, not a reference to the step itself? Could this run us into trouble? Maybe for now we can name this `getStepClone()` to make it clear that if the step changes in the future, we still only have a copy? I could imagine that a step would be modified between the clone and the eventual usage, right? That's OK for many uses, but just needs to be clearly indicated, if my understanding of `cloneDeep` is right.

Reply:
But if we can pass the reference to the clone, why can't we pass the reference to the step itself? 

For ref: https://lodash.com/docs/4.17.10#cloneDeep

Reply:
@jywarren Actually I though that modifying this returned object should not affect another step, hence I used cloneDeep(), well It's just a matter of choice. Anyway I'll change this to return the reference😁

Reply:
Oh cool then for time being we can probably remove lodash!

Reply:
But we may want to know not the output but the configuration of a future module. I just think it's hard to predict what'll be useful, so why restrict now? :+1: